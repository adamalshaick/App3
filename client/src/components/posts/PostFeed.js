import React, { Component } from "react";
import PropTypes from "prop-types";
import PostItem from "./PostItem";

class PostFeed extends Component {
  render() {
    const { posts, profiles, profile } = this.props;
    const postFeed = posts.map(post => (
      <PostItem key={post._id} post={post} profile={profile} />
    ));
    return <div className="col-12">{postFeed}</div>;
  }
}

PostFeed.propTypes = {
  posts: PropTypes.array.isRequired
};

export default PostFeed;
