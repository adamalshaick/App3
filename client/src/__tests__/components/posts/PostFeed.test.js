import React from "react";
import PostFeed from "../../../components/posts/PostFeed";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

let wrapper;

const Props = {
  posts: [],
  profile: {}
};

wrapper = shallow(<PostFeed {...Props} />);

describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
