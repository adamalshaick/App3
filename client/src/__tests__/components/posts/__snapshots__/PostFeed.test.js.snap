// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<div
  className="col-12"
/>
`;
